### PR TITLE
GEN-1570 - show reviews distribution

### PR DESCRIPTION
**Pull request by guilhermespopolin**
## Describe your changes

* Add `getReviewsDistribution` utility function: it returns the distribution of reviews by score in a form of a rounded percentage using [_Largest Remainder Method_](https://stackoverflow.com/questions/13483430/how-to-make-rounded-percentages-add-up-to-100).
* Add `ReviewDistribution` component that displays the the reviews distribution by score.
* Some refactorings:
  * Move `@/components/Stars/Stars.tsx` --> `@/components/ProductReviews/Stars.tsx` and update import statements accordantly 
  * Move+Rename `@/components/RatingCount/RatingCount.tsx` --> `@/components/ProductReviews/ReviewsDistributionByScore.tsx` and update import statements accordantly.
  * Update `Stars` component API: `score` prop is expected to be a string instead of a number. Down in the stack it's going to assume a specific type: `type Score = '5' | '4' | '3' | '2' | '1'`
  * Change `getProductReviewComments` return value;

Screenshot 2023-12-14 at 15 40 19

Can be tested [here](https://hedvig-dot-com-git-gen-1570-refactreviews-distribution-hedvig.vercel.app/se/forsakringar/olycksfallsforsakring/av)

## Justify why they are needed

Part of _Product Reviews_ project.

### CODE REVIEW

**Comment by guilhermespopolin** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #3751** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3751?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #3752** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3752?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
      * **PR #3753** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3753?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3753?utm_source=stack-comment).

File: apps/store/src/services/productReviews/getReviewsDistribution.test.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { describe, expect, test } from '@jest/globals'
+import { getReviewsDistribution } from './getReviewsDistribution'
+import { type ReviewComments } from './productReviews.types'
+
+describe('getReviewsDistribution', () => {
```
Comment (alebedev) [inline apps/store/src/services/productReviews/getReviewsDistribution.test.ts:5]: Discussion topic - do you see value in singe top-level `describe()` ?  I think filename provides enough context and less nesting makes it easier to work with the code

File: apps/store/src/services/productReviews/getReviewsDistribution.test.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+    // Act
+    const reviewsDistribution = getReviewsDistribution(reviews)
+    const distributionSum = reviewsDistribution.reduce(
+      (acc, [, percentage]) => acc + Number(percentage),
+      0,
+    )
+
+    // Assert
+    expect(distributionSum).toEqual(100)
```
Comment (alebedev) [inline apps/store/src/services/productReviews/getReviewsDistribution.test.ts:26]: Would be nice to assert specific results too

**Review by alebedev — APPROVED** [review]
Great work with implementing and testing the algorithm!

File: apps/store/src/services/productReviews/getReviewsDistribution.test.ts (hunk 2 above)
Comment (guilhermespopolin) [reply apps/store/src/services/productReviews/getReviewsDistribution.test.ts:26]: Yes, though the same. Gonna update it.

File: apps/store/src/services/productReviews/getReviewsDistribution.test.ts (hunk 1 above)
Comment (guilhermespopolin) [reply apps/store/src/services/productReviews/getReviewsDistribution.test.ts:5]: Good question. Initially I though that single top-level `describe` would at least log results in a clearer format but it seems not to be the case:

**Without single top-level `describe`**

Screenshot 2023-12-19 at 12 54 35

**With single top-level `describe`**

Screenshot 2023-12-19 at 12 54 56

So I don't have arguments against removing it.

**Comment by guilhermespopolin** [comment]
### Merge activity

* **Dec 20, 3:28 AM**: @@guilhermespopolin started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/3753).

**Pull request merged**